lib/tool_belt/repository.rb

```diff
@@ -0,0 +1,50 @@
+require 'git'
+
+module ToolBelt
+  class Repository
+
+    attr_reader :name, :branch, :url, :version_branch, :main, :options
+
+    def initialize(name, args = {}, options = {})
+      @name = name
+      @branch = args[:branch]
+      @url = args[:repo]
+      @version_branch = args[:version_branch]
+      @main = args[:main]
+      @options = options
+    end
+
+    def self.parse(config)
+      config.options.repos.map do |repo|
+        self.new(*repo, namespace: config.options.namespace)
+      end
+    end
+
+    def branch!
```

Comment:
Why an exclamation method here?

Reply:
Mostly because it does potentially scary things like pushing a new branch upstream.